modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataOutputDelegate.java

```diff
@@ -188,7 +188,7 @@ private MetadataType adaptToListIfNecessary(MetadataType resolvedType, Object ke
       resolvedType = wrapInMessageType(resolvedType, key, metadataContext, message.getAttributesType());
     }
 
-    return metadataContext.getTypeBuilder().arrayType().id(typeId).of(resolvedType).build();
+    return metadataContext.getTypeBuilder().arrayType().of(resolvedType).build();
```

Comment:
have you run all test cases to verify that the SDK doesn't try to obtain the list type?